scopes/harmony/application/application.main.runtime.ts

```diff
@@ -97,6 +97,15 @@ export class ApplicationMain {
     return this.appSlot.get(id.toString());
   }
 
+  /**
+   * get an application by a component id.
+   */
+  getAppById(id: ComponentID) {
+    const apps = this.listAppsById(id);
+    if (!apps) return undefined;
+    return head(apps);
```

Comment:
Why only head? what if I many apps in the same component?
list them all

Reply:
there is an issue there where we support a single application but we have a listAppById API. see registerApp API. @davidfirst also asked if we are supporting multiple apps or not per component and decided to go against. For now, left head but in cannot happen practically since register API doesn't support. and we need to remove over time